README.md

```diff
@@ -1,69 +1,103 @@
+| [theiaide/theia-swift](https://hub.docker.com/r/theiaide/theia-swift) | Theia-based Swift example application | |
+| [theiaide/theia](https://hub.docker.com/r/theiaide/theia) | Theia-based JavaScript/TypeScript (Web Technologies) example application | |
 
-[![dockeri.co](http://dockeri.co/image/theiaide/theia)](https://hub.docker.com/r/theiaide/theia/)
+---
 
-`theiaide/theia` image contains an example of Theia based IDE for Web Developers.
+<br />
 
-### How to use `theiaide/theia` image?
+### Electron Apps
 
-At the moment Theia is still in [active development](https://github.com/theia-ide/theia#roadmap). It is recommended to use [`theiaide/theia:next`](#typefoxtheianext) image to have a look at the current state.
+<img
+    src='./assets/electron.png'
+    alt='electron logo'
+    width="50px"
+/>
 
-This script pulls the image and runs Theia IDE on http://localhost:3000 with the current directory as a workspace. The option of `--init` is added to fix the [defunct process problem](https://github.com/theia-ide/theia-apps/issues/195).
+| Application Name | Description | Documentation |
+|:---|:---|:---|
+| `theia-cpp-electron` | Theia-based C/C++ desktop IDE | [docs](./theia-cpp-electron/README.md)
 
-    # Linux or macOS
-    docker run -it --init -p 3000:3000 -v "$(pwd):/home/project:cached" theiaide/theia:next
-    
-    # Windows
-    docker run -it --init -p 3000:3000 -v "%cd%:/home/project:cached" theiaide/theia:next
+---
 
+<br />
 
-You can pass additional arguments to Theia after the image name, for example to enable debugging:
+### Building & Running
 
-    # Linux or macOS
-    docker run -it --init -p 3000:3000 --expose 9229 -p 9229:9229 -v "$(pwd):/home/project:cached" theiaide/theia:next --inspect=0.0.0.0:9229
-    
-    # Windows
-    docker run -it --init -p 3000:3000 --expose 9229 -p 9229:9229 -v "%cd%:/home/project:cached" theiaide/theia:next --inspect=0.0.0.0:9229
+The following will use the `theiaide/theia` image as an example:
 
-### Image Variants
+1. Build the `theiaide/theia` image:
 
-#### `theiaide/theia:latest`
+    `theiaide/theia:latest` - build the latest version:
 
-This image is based on the latest stable released version.
+    ```bash
+    $> docker build --no-cache --build-arg version=latest --build-arg strip=true -t theia:latest .
+    ```
 
-#### `theiaide/theia:<version>`
+    `theiaide/theia:next` - build the next version:
 
-This image is based on the given stable released version.
+    ```bash
+    $> docker build --no-cache --build-arg version=next --build-arg strip=true -t theia:next .
+    ```
 
-#### `theiaide/theia:next`
+2. Run the image:
 
-This image is based on the nightly published version.
+    ```bash
+    $> docker run --security-opt seccomp=unconfined --init -it -p 3000:3000 theiaide/theia:next
+    ```
 
-#### `theiaide/theia:<version>-next.<commit hash>`
+<br />
 
-This image is based on the given nightly published version.
+## Tips & Tricks
 
```

Comment:
Do you have any Tips & Ticks that we should mentioned here

Reply:
I was hoping others might have some suggestions, perhaps @marcdumais-work regarding some gotchas with permissions, or adding debug symbols.

Reply:
What about a suggestion to verify which **nodejs** and **yarn** being installed on your computer this could be a way to see why the applications does not compile. This is just a suggestion here

Reply:
> What about a suggestion to verify which **nodejs** and **yarn** being installed on your computer this could be a way to see why the applications does not compile. This is just a suggestion here

The node and yarn versions are handled through the Dockerfile for each image.
I don't believe it makes sense to display in the root readme.

Reply:
Also, since theia works as a server-client, if we gave them and example to start Theia (Docker images  run...) which is the server side and how to connect using their web browser (client side)